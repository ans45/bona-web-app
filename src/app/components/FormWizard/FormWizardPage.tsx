/**
 *
 * FormWizardPage
 *
 */
import React, { ReactNode } from 'react';
import { ValidationErrors } from 'final-form';

export interface Props {
  children?: ReactNode;
  validate: (
    Object,
  ) => ValidationErrors | Promise<ValidationErrors> | undefined;
}

export const FormWizardPage = ({ children }: Props) => {
  return <>{children}</>;
};
