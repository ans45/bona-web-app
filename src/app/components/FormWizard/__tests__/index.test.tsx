import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';

import { FormWizard } from '..';
import { FormWizardPage } from '../FormWizardPage';

describe('<FormWizard  />', () => {
  const mockValidate = jest.fn();
  const mockHandleSubmit = jest.fn();
  const mockValues = {};

  beforeEach(() => {
    mockHandleSubmit.mockReset();
  });

  it('should match snapshot', () => {
    const loadingIndicator = render(
      <FormWizard initialValues={mockValues} onSubmit={mockHandleSubmit}>
        <FormWizardPage validate={mockValidate} />
        <FormWizardPage validate={mockValidate} />
      </FormWizard>,
    );
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });

  it('should show previous button if past the first child page', () => {
    render(
      <FormWizard initialValues={mockValues} onSubmit={mockHandleSubmit}>
        <FormWizardPage validate={mockValidate} />
        <FormWizardPage validate={mockValidate} />
      </FormWizard>,
    );

    const nextButton = screen.getByTestId('wizard-next');
    fireEvent.click(nextButton);

    expect(screen.getByTestId('wizard-previous')).toBeVisible();
  });

  it('should show next button if not on the last page', () => {
    render(
      <FormWizard initialValues={mockValues} onSubmit={mockHandleSubmit}>
        <FormWizardPage validate={mockValidate} />
        <FormWizardPage validate={mockValidate} />
      </FormWizard>,
    );
    const nextButton = screen.getByTestId('wizard-next');
    expect(nextButton).toBeVisible();
  });

  it('should show submit button if on the last page', () => {
    render(
      <FormWizard initialValues={mockValues} onSubmit={mockHandleSubmit}>
        <FormWizardPage validate={mockValidate} />
        <FormWizardPage validate={mockValidate} />
      </FormWizard>,
    );

    const nextButton = screen.getByTestId('wizard-next');
    fireEvent.click(nextButton);

    expect(screen.getByTestId('wizard-submit')).toBeVisible();
  });

  it('should be able to change pages', () => {
    render(
      <FormWizard initialValues={mockValues} onSubmit={mockHandleSubmit}>
        <FormWizardPage validate={mockValidate}>
          <h1>Page 1</h1>
        </FormWizardPage>
        <FormWizardPage validate={mockValidate}>
          <h1>Page 2</h1>
        </FormWizardPage>
      </FormWizard>,
    );

    expect(screen.getByText('Page 1')).toBeVisible();

    const nextButton = screen.getByTestId('wizard-next');
    fireEvent.click(nextButton);

    expect(screen.getByText('Page 2')).toBeVisible();

    const previousButton = screen.getByTestId('wizard-previous');
    fireEvent.click(previousButton);

    expect(screen.getByText('Page 1')).toBeVisible();
  });
});
