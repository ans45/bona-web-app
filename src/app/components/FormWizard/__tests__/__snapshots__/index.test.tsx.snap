// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormWizard  /> should match snapshot 1`] = `
<form>
  <div
    class="buttons"
  >
    <button
      data-testid="wizard-next"
      type="submit"
    >
      Next Â»
    </button>
  </div>
</form>
`;
